source/python/idetect/nlp_models/base_model.py

```diff
@@ -1,11 +1,18 @@
 import errno
 import fcntl
 import os
-
+import re
+import numpy as np
 import pandas as pd
 import requests
+import spacy
 from sklearn.externals import joblib
+from sklearn.base import TransformerMixin, BaseEstimator
+from scipy import sparse
+from gensim import matutils, models
+from gensim.sklearn_integration.sklearn_wrapper_gensim_lsimodel import SklLsiModel
 
+from idetect.geotagger import strip_accents, compare_strings, strip_words, common_names, LocationType, subdivision_country_code, match_country_name, city_subdivision_country
 
```

Comment:
I think there are a few unused imports, such as `spacy` and all the stuff from `idetect.geotagger`